### ISSUE TITLE
DuplexStream?

### ISSUE THREAD

**Issue by moll** [rkusa/frame-stream#3]
Hey,

Have you thought of making FrameStream into a duplex stream so one could keep a reference to single stream to read and write from? E.g. wrap an existing `Net.Socket` transparently with a `FrameStream` rather than having to pass around one `FrameSteam` and one `LengthPrefixStream`.

Thanks

**Comment by moll** [comment]
Okay, I now realize a Transform is in fact a duplex stream. I'm probably thinking of something like QuadruplexStream (^_^) that passes its reads through one transform and writes through another. Perhaps there's a module out there that does it, but NPM.org-s search is just so poor that I haven't been able to find it.

**Comment by rkusa** [comment]
I'd also prefer to have only one stream instead of two, but I am not aware of something like a QuadruplexStream ;-)

**Comment by moll** [comment]
I managed to find https://github.com/Raynos/duplexer yesterday that seems like a way to combine to transforms to a single stream. It does however do some weird function overwriting as opposed to just creating an intermediate duplex stream. Don't know what's the benefit of that. Perhaps not having another buffer in between.

I guess it's safe to say this need not be the responsibility of FrameStream now that there is a combine function. I'll close this issue then.

PS. On a related topic, the naming of `FrameStream` and `LengthPrefixStream` could be improved given that they're basically the same coin from both sides. Currently they're are a little ambiguous. I named them `FrameEncoder` and `FrameDecoder` in my code.
